src/taskgraph/transforms/from_deps.py

```diff
@@ -177,14 +201,40 @@ def from_deps(config, tasks):
 
             primary_dep = [dep for dep in group if dep.kind == primary_kind][0]
 
-            if primary_dep.label.startswith(primary_kind):
-                new_task["name"] = primary_dep.label[len(primary_kind) + 1 :]
-            else:
-                new_task["name"] = primary_dep.label
+            if set_name:
+                if primary_dep.label.startswith(primary_kind):
+                    new_task["name"] = primary_dep.label[len(primary_kind) + 1 :]
+                else:
+                    new_task["name"] = primary_dep.label
 
             if copy_attributes:
-                attrs = new_task.get("attributes", {})
+                new_task.setdefault("attributes", {})
+                attrs = new_task["attributes"]
                 new_task["attributes"] = primary_dep.attributes.copy()
                 new_task["attributes"].update(attrs)
 
+            if fetches:
+                new_task.setdefault("fetches", {})
+                task_fetches = new_task["fetches"]
+
+                for dep_label in new_task["dependencies"]:
+                    dep_task = config.kind_dependencies_tasks[dep_label]
```

Comment:
I almost didn't bother commenting because it's such a trivial thing.. but you could use [get_dependencies](https://github.com/taskcluster/taskgraph/blob/main/src/taskgraph/util/dependencies.py#L50) at this point.